### ISSUE TITLE
Coverage submission failing counts as a test failure

### ISSUE THREAD

**Issue by jackfirth** [rpless/cover-coveralls#4]
With the current example Travis script, submitting test coverage counts as part of the testing commands. This means if the tests pass but coverage submission fails (for example due to network issues or Coveralls setup issues), the build fails with a test failure state.

I think most would prefer if the coverage process was independent of the testing process, but submitted only when tests pass. This can be done in travis by moving the coverage command from the `script:` block to an `after_success:` block (see the [Travis Build Lifecycle](http://docs.travis-ci.com/user/customizing-the-build/#The-Build-Lifecycle) docs for details):

```yml
script:
 - raco test $TRAVIS_BUILD_DIR # run tests. you wrote tests, right?

after_success:
 - raco cover -f coveralls -d $TRAVIS_BUILD_DIR/coverage . # generate coverage information for coveralls
```

Some people *want* coverage submission failing to fail the build, but I think most don't and thus this makes for a better default.

**Comment by florence** [comment]
@rpless seems reasonable. Do you agree?

**Comment by rpless** [comment]
Yes, I'll make changes to the README to show how to do it both ways. 

This is a really good call out, especially in light of the fact that coveralls isn't really a great service. We had a ton of trouble with it on [Finch](https://github.com/finagle/finch). Builds failed constantly and then we made the coveralls task run in `after_success` we would often find that we didn't have any coverage info. In my mind there are better options now (as opposed to when we wrote cover). I'm hoping to implement a format for one of these services within the week.
